pallets/phala/src/compute/wrapped_balances.rs

```diff
@@ -133,6 +144,225 @@ pub mod pallet {
+				T::WPhaAssetId::get(),
+			)
+		}
+
+		fn deactivate(_amount: Self::Balance) {
+			unimplemented!()
+		}
+
+		fn reactivate(_amount: Self::Balance) {
+			unimplemented!()
+		}
+
+		fn minimum_balance() -> Self::Balance {
+			<pallet_assets::pallet::Pallet<T> as Inspect<T::AccountId>>::minimum_balance(
+				T::WPhaAssetId::get(),
+			)
+		}
+
+		fn burn(mut _amount: Self::Balance) -> Self::PositiveImbalance {
+			unimplemented!()
+		}
+
+		fn issue(mut _amount: Self::Balance) -> Self::NegativeImbalance {
+			unimplemented!()
+		}
+
+		fn ensure_can_withdraw(
+			who: &T::AccountId,
+			amount: Self::Balance,
+			_reasons: WithdrawReasons,
+			new_balance: Self::Balance,
+		) -> DispatchResult {
+			if amount.is_zero() {
+				return Ok(());
+			}
+			let lock = Self::get_election_lock(who);
+			ensure!(new_balance >= lock, Error::<T>::LiquidityRestrictions);
+			Ok(())
+		}
+
+		fn transfer(
+			_transactor: &T::AccountId,
+			_dest: &T::AccountId,
+			_value: Self::Balance,
+			_existence_requirement: ExistenceRequirement,
+		) -> DispatchResult {
+			unimplemented!()
+		}
+
+		fn slash(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+		) -> (Self::NegativeImbalance, Self::Balance) {
+			unimplemented!()
+		}
+
+		fn deposit_into_existing(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+		) -> Result<Self::PositiveImbalance, DispatchError> {
+			unimplemented!()
+		}
+
+		fn deposit_creating(_who: &T::AccountId, _value: Self::Balance) -> Self::PositiveImbalance {
+			unimplemented!()
+		}
+
+		fn withdraw(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+			_reasons: WithdrawReasons,
+			_liveness: ExistenceRequirement,
+		) -> Result<Self::NegativeImbalance, DispatchError> {
+			unimplemented!()
+		}
+
+		fn make_free_balance_be(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+		) -> SignedImbalance<Self::Balance, Self::PositiveImbalance> {
+			unimplemented!()
+		}
+	}
+
+	impl<T: Config> ReservableCurrency<T::AccountId> for Pallet<T>
+	where
+		BalanceOf<T>: sp_runtime::traits::AtLeast32BitUnsigned + Copy + FixedPointConvert + Display,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + pallet_balances::Config + vault::Config + pallet_elections_phragmen::Config,
+		T: pallet_balances::Config<Balance = BalanceOf<T>>,
+	{
+		fn can_reserve(who: &T::AccountId, value: Self::Balance) -> bool {
+			Self::total_balance(who) >= value
+		}
+		fn reserved_balance(who: &T::AccountId) -> Self::Balance {
+			ElectionReserves::<T>::get(who).unwrap_or_default()
+		}
+		fn reserve(who: &T::AccountId, value: Self::Balance) -> DispatchResult {
+			let actual = value.min(Self::total_balance(who));
+			ElectionReserves::<T>::mutate(who, |reserve| {
+				*reserve = Some(reserve.unwrap_or_default() + actual);
+			});
+			Ok(())
+		}
+		fn unreserve(who: &T::AccountId, value: Self::Balance) -> Self::Balance {
+			let actual = value.min(ElectionReserves::<T>::get(who).unwrap_or_default());
```

Comment:
You can set `ElectionReserves` as `ValueQuery`. So you don't need to call `unwrap_or_default()` every time.

Reply:
done